gensim/utils.py

```diff
@@ -376,11 +376,26 @@ def _save_specials(self, fname, separately, sep_limit, ignore, pickle_protocol,
                 elif isinstance(val, sparse_matrices) and val.nnz >= sep_limit:
                     separately.append(attrib)
 
+        def delete_attribute(object, attribute):
+            if hasattr(object, attribute):
+                asides[attribute] = getattr(object, attribute)
+                delattr(object, attribute)
+                
         # whatever's in `separately` or `ignore` at this point won't get pickled
         for attrib in separately + list(ignore):
-            if hasattr(self, attrib):
-                asides[attrib] = getattr(self, attrib)
-                delattr(self, attrib)
+            # As a result of maintaining backwards compatibility through __getattr__ after refactoring
+            # "syn0", "syn0norm", "vocab", and "index2word" out of Word2Vec to KeyedVectors,
+            # hasattr(self, "syn0") will return True but delattr(self, "syn0") will fail because it's
+            # not a true attribute of Word2Vec, Word2Vec.__getattr__ just reroutes it to KeyedVectors' attributes
+            if attrib in ["syn0", "syn0norm", "vocab", "index2word"]:
+                if type(self) == "Word2Vec":  # if saving a Word2Vec model, syn0, etc. are stored in self.kv
```

Comment:
This currently doesn't work as intended because `type(self)` returns a class, not a string. But even if changed to `type(self).__name__`, this would call `delete_attribute` with the Word2Vec object and syn0/syn0norm, and so when properties are being restored, syn0/syn0norm would be restored to the word2vec object instead of the KeyedVecs object. 

The correct way to handle it seems via the recursive_saveloads that is already implemented, but we'll still need a special condition check to ensure delete_attribute isn't called on the word2vec object. 

All this need for special condition checks is resulting from the getattr patching to handle syn0, vocab and syn0norm backwards compatibility. 

Even if we decide not to maintain syn0 and syn0norm backwards compatibility, vocab is definitely part of the API, and there doesn't seem to be any way to avoid special patching.